src/core/pb_hkdf_sha256.rs

```diff
@@ -17,6 +17,21 @@ use k256::sha2::Sha256;
 #[display("{}", self.description())]
 pub struct PbHkdfSha256 {}
 
+impl PbHkdfSha256 {
+    pub fn hkdf_key_agreement(
+        ikm: impl AsRef<[u8]>,
+        salt: Option<&[u8]>,
+        info: Option<&[u8]>,
+    ) -> Exactly32Bytes {
```

Comment:
Ah, you can change: ` salt: Option<&[u8]>,` -> ` salt: impl Into<Option<&[u8]>>,` which is nice because then you dont need to write `Some(salt)` later when you use it, you can simply pass `salt` because `T` implements `Into<Option<T>>` by Rust :)

Same for `info` ofc...

Reply:
I mostly didn't want to add it as it requires the addition of lifetime, but fair enough.
so this function becomes:
```
pub fn derive_key<'a>(
        ikm: impl AsRef<[u8]>,
        salt: impl Into<Option<&'a [u8]>>,
        info: impl Into<Option<&'a [u8]>>,
) -> Exactly32Bytes {
```

Reply:
I wonder if:

```rust
pub fn derive_key<'a>(
        ikm: impl AsRef<[u8]>,
        salt: impl Into<Option<impl AsRef<[u8]>>>,
        info: impl Into<Option<impl AsRef<[u8]>>>,
) -> Exactly32Bytes {
```

works?

Reply:
no,
but still, it is ok with lifetime annotation I posted above.